### PR TITLE
docs(command): zebra download

### LINKED ISSUE #3383: Document `zebrad download` command
Zebra has a command to pre download the parameters. Zebra also download parameters at `zebra start` if they are not present.

There is no documentation about this, i think we should add:

- Briefly describe what are this parameters and why they are needed.
- Document the `zebrad download` existence.
- Explain `zebrad start` will automatically start downloading parameters at startup.

https://zebra.zfnd.org/user/run.html

We may need to wait on https://github.com/ZcashFoundation/zebra/issues/3133 as some of the current behaviour will maybe change.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

We want to have some sort of the documentation for the zk-SNARK parameters zebra download at startup or using the `zebrad download` command.

Close https://github.com/ZcashFoundation/zebra/issues/3383

## Solution

- Add a new section to the book.
- Document the parameters and their usage in zebra.
- Add an extra logging line to show the process of downloading/verifying parameters is complete.

## Review

Anyone can review.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: book/src/user/parameters.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+$
+```
+
+## zebrad start
+
+The alternative way is to let zebra do the above process at startup.
+
+Before Zebra attempts to verify any of the 3 mentioned circuits it needs to have the parameters in place. For this reason zebra start will check for the parameters and download them if needed each time it is started.
+
+At zebra startup, when initializing the verifiers, a separate task is created to do the same as the `zebra download` command. This allows Zebra sync to make progress before having the parameters. Note that these parameters are needed after Sapling activation which happens at block `419_200` in the Mainnet and block `15` in the Testnet. Zebra will panic if the parameters are not there when verifying those blocks and above.
```
Comment (teor2345) [inline book/src/user/parameters.md]: Can you show me where this panic happens?

I tried to design it so Zebra would wait for the download to finish, but I might have got it wrong.

```suggestion
At zebra startup, when initializing the verifiers, a separate task is created to do the same as the `zebra download` command. This allows Zebra sync to make progress before having the parameters. Note that these parameters are needed after Sapling activation which happens at block `419_200` in the Mainnet and block `15` in the Testnet. Zebra will wait for the parameters to download before verifying those blocks and above.
```

File: book/src/user/parameters.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+There are 2 different zebrad commands to get this parameters from the internet and load them into zebra:
+
+## zebrad download
+
+When this command is executed Zebra will create a path for each parameter file. For example, in linux, parameter files will be at:
+
+- `/home/$USER/.zcash-params/sapling-output.params`
+- `/home/$USER/.zcash-params/sapling-spend.params`
+- `/home/$USER/.zcash-params/sprout-groth16.params`
+
```
Comment (teor2345) [inline book/src/user/parameters.md:48]:
```suggestion

These are the same parameter paths used by `zcashd` and [fetch-params.sh](https://github.com/zcash/zcash/blob/master/zcutil/fetch-params.sh).

```

File: zebra-consensus/src/primitives/groth16/params.rs [hunk 3]
```diff
@@ -101,6 +101,8 @@ impl Groth16Parameters {
                 .expect("unreachable code: sprout loader panics on failure"),
         };
 
+        tracing::info!("Zcash Sapling and Sprout parameters downloaded and/or verified");
```
Comment (teor2345) [inline zebra-consensus/src/primitives/groth16/params.rs:104]: Optional: We already log when the parameters are downloaded in `zebra-consensus`. Should we delete the `zebra-consensus` log as part of this change?

`zebra-consensus` only logs when it is run by `zebrad start`, so I think this log is better.

https://github.com/ZcashFoundation/zebra/blob/f3f8254e4ad84c14fb3f0cbd4386df4ef4795fb0/zebra-consensus/src/chain.rs#L248

**Review by teor2345 — COMMENTED** [review]
Looks good, I just want to check about the panic, that seems bad.

File: book/src/user/parameters.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# Zebra zk-SNARK Parameters
+
+The privacy features provided by Zcash are backed by different [zk-snarks proving systems](https://z.cash/technology/zksnarks/) which are basically cryptographic primitives that allow a prover convince a verifier that a statement is true revealing no more information than the proof itself.
```
Comment (mpguerra) [inline book/src/user/parameters.md]:
```suggestion
The privacy features provided by Zcash are backed by different [zk-snarks proving systems](https://z.cash/technology/zksnarks/) which are basically cryptographic primitives that allow a prover to convince a verifier that a statement is true by revealing no more information than the proof itself.
```

File: book/src/user/parameters.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+One of this proving systems is [Groth16](https://eprint.iacr.org/2016/260.pdf) and it is the one used by the Zcash transactions version 4 and greater. More specifically, in the sapling spend/output descriptions circuits and in the sprout joinsplits descriptions circuit.
```
Comment (mpguerra) [inline book/src/user/parameters.md]:
```suggestion
One of these proving systems is [Groth16](https://eprint.iacr.org/2016/260.pdf) and it is the one used by the Zcash transactions version 4 and greater. More specifically, in the sapling spend/output descriptions circuits and in the sprout joinsplits descriptions circuit.
```

File: book/src/user/parameters.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+The Groth16 proving system requires a trusted setup, this is a set of predefined parameters that every node should possess to verify the proofs that will show up in the blockchain.
+
+These parameters are in the form of files, they are basically predefined keys that will allow verification of the circuits. They were initially obtained by [this process](https://eprint.iacr.org/2017/1050.pdf).
+
+3 files are needed, one for each circuit, this is part of the Zcash consensus protocol:
+
+https://zips.z.cash/protocol/protocol.pdf#grothparameters
+
+In Zebra, this 3 public files are mapped into 2 structures:
```
Comment (mpguerra) [inline book/src/user/parameters.md]:
```suggestion
In Zebra, these 3 public files are mapped into 2 structures:
```

File: book/src/user/parameters.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+pub struct SproutParameters {
+    pub joinsplit_prepared_verifying_key: groth16::PreparedVerifyingKey<Bls12>,
+}
+```
+
+Zebra uses the [bellman crate groth16 implementation](https://github.com/zkcrypto/bellman/blob/main/src/groth16/mod.rs) for all groth16 types.
+
+Each time a transaction has any sprout joinsplit, sapling spend or sapling output these loaded parameters will be used for the verification process. Zebra verifies in parallel and by batches, these parameters are used on each verification done.
+
+There are 2 different zebrad commands to get this parameters from the internet and load them into zebra:
```
Comment (mpguerra) [inline book/src/user/parameters.md]:
```suggestion
There are 2 different zebrad commands to get these parameters from the internet and load them into zebra:
```

File: book/src/user/parameters.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+## zebrad download
+
+When this command is executed Zebra will create a path for each parameter file. For example, in linux, parameter files will be at:
+
+- `/home/$USER/.zcash-params/sapling-output.params`
+- `/home/$USER/.zcash-params/sapling-spend.params`
+- `/home/$USER/.zcash-params/sprout-groth16.params`
+
+This files are available for [download](https://download.z.cash/downloads/) and their hash for verification are part of the Zcash protocol:
```
Comment (mpguerra) [inline book/src/user/parameters.md]:
```suggestion
These files are available for [download](https://download.z.cash/downloads/) and their hash for verification is part of the Zcash protocol: 
```

**Review by mpguerra — DISMISSED** [review]
Looks good, great summary! A few typos to fix and that's it from me

File: zebra-consensus/src/primitives/groth16/params.rs (hunk 3 above)
Comment (oxarbitrage) [reply zebra-consensus/src/primitives/groth16/params.rs:104]: https://github.com/ZcashFoundation/zebra/pull/5901/commits/db847d67adf569a4374a4bfa740d9fddc5ab9154

**Review by teor2345 — APPROVED** [review]
Thanks, looks good to me!

**Pull request merged**